File: frontend/src/types/scan.ts
export interface Scan {
  id string;
  filename string;
  source_tool string;
  upload_date string;
  total_findings number;
  critical_count number;
  high_count number;
  medium_count number;
  low_count number;
  info_count number;
  processed boolean;
  processing_error string;
}

File: backend/app/models/scan.py
from sqlalchemy import Column, String, Integer, Boolean, DateTime, Text
from sqlalchemy.dialects.postgresql import UUID
from datetime import datetime
import uuid
from app.database import Base

class Scan(Base):
    __tablename__ = "scans"

    id = Column(UUID(as_uuid=True), primary_key=True, default=uuid.uuid4)
    filename = Column(String(255), nullable=False)
    source_tool = Column(String(50), nullable=False)
    upload_date = Column(DateTime, default=datetime.utcnow)
    total_findings = Column(Integer, default=0)
    critical_count = Column(Integer, default=0)
    high_count = Column(Integer, default=0)
    medium_count = Column(Integer, default=0)
    low_count = Column(Integer, default=0)
    info_count = Column(Integer, default=0)
    processed = Column(Boolean, default=False)
    processing_error = Column(Text, nullable=True)